src/js/stores/AppsStore.js

```diff
@@ -147,29 +148,47 @@ function processApp(app) {
   return app;
 }
 
-function processEmptyGroup(group) {
-  return Object.assign(processApp({id: group.id}), {
-    isGroup: true,
-    status: null
+function addData(parent, child) {
+  Object.keys(parent).forEach(key => {
+    if (key === "health") {
+      parent[key].forEach((value, index) => {
+        value.quantity += child[key][index].quantity;
+      });
+      return;
+    }
+    parent[key] += child[key];
   });
 }
 
 // TODO: refactor to adhere to /groups endpoint structures.
 // https://github.com/mesosphere/marathon/issues/3383
-function processAppsAndEmptyGroups(group, apps = []) {
-  if (group.apps != null && group.apps.length > 0) {
-    group.apps.forEach(app => {
-      apps.push(processApp(app));
-    });
-  } else if (group.id !== "/") {
-    apps.push(processEmptyGroup(group));
+function processAppsAndGroups(group, root = true) {
```

Comment:
This aggregates also the totalCpu etc. for a group so this data has to be passed up in the recursive function, on root level we only want to have the flat list. Also we do not want to add the root group to the flat list. Please keep in mind that we find the deepest leave and start then to work our way back up in the recursive function. So we need to find the applications and then sum all the data for the groups upwards. The root is the root of the tree structure so this does not need to be `id: "/"` so the parameter could also be named `onlyProvideFlatList`